array.c

```diff
@@ -3066,11 +3095,21 @@ rb_ary_to_s(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.to_a     -> ary
+ *    to_a -> self or new_array
  *
- *  Returns +self+.
+ *  Returns +self+ if <tt>self.instance_of?(Array)</tt>:
+ *    a = [:foo, 'bar', 2]
+ *    a1 = a.to_a
+ *    a1 # => [:foo, "bar", 2]
+ *    a1.equal?(a) # => true # Returned self
  *
- *  If called on a subclass of Array, converts the receiver to an Array object.
+ *  If +self+ is a subclass of \Array, returns the new \Array
+ *  formed by converting +self+ to an \Array:
+ *    class MyArray < Array; end
+ *    a = MyArray.new([:foo, 'bar', 2])
```

Comment:
This is one of the biggest gotchas when subclassing array, and consider my recommendation as a nitpick: 

```suggestion
 *    a = MyArray.new([:foo, 'bar', 2])
 *    a.class   #=> MyArray
```

I'm commenting this because in places as `irb` or when doing `puts` debugging the `a` variable looks like an array, but it's not.

Reply:
Thanks, @esparta.  I thought of that, but did not do it (but should have).  Will fix.